src/main/java/frc/robot/subsystems/Arm.java

```diff
@@ -0,0 +1,118 @@
+
+    private static Arm currentInstance;
+    // TODO Change placeholder value to robotSpecific
+    WPI_TalonSRX armTalon;
+    ErrorCode errorCode;
+
+    private Arm() {
+        // Init all the talon values
+        armTalon = new WPI_TalonSRX(Config.ARM_TALON);
+
+        // Config factory default to clear out any lingering values
+        armTalon.configFactoryDefault();
+
+        // Allow the arm to be moved easily when disabled
+        armTalon.setNeutralMode(NeutralMode.Coast);
+
+        // Setup the talon, recording the error code
+        errorCode = armTalon.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative,
+                0, Config.CAN_TIMEOUT_SHORT);
+
+        armTalon.setSelectedSensorPosition(0);
+
+        armTalon.setInverted(Config.INVERT_ARM_TALON);
+
+        /* Config the peak and nominal outputs, 12V means full */
+        armTalon.configNominalOutputForward(0, Config.CAN_TIMEOUT_SHORT);
+        armTalon.configNominalOutputReverse(0, Config.CAN_TIMEOUT_SHORT);
+        armTalon.configPeakOutputForward(1, Config.CAN_TIMEOUT_SHORT);
+        armTalon.configPeakOutputReverse(-1, Config.CAN_TIMEOUT_SHORT);
+
+        armTalon.configAllowableClosedloopError(0, Config.ARM_ALLOWABLE_CLOSED, Config.CAN_TIMEOUT_SHORT);
+
+        //  Config the PID Values based on constants
+        armTalon.config_kP(0, Config.ARM_P, Config.CAN_TIMEOUT_SHORT);
+        armTalon.config_kI(0, Config.ARM_I, Config.CAN_TIMEOUT_SHORT);
+        armTalon.config_kD(0, Config.ARM_D, Config.CAN_TIMEOUT_SHORT);
+        armTalon.config_kF(0, Config.ARM_F, Config.CAN_TIMEOUT_SHORT);
+
+        // Set up the close loop period
+        armTalon.configClosedLoopPeriod(0, Config.CAN_TIMEOUT_LONG);
+        armTalon.setSensorPhase(true);
+        armTalon.setStatusFramePeriod(StatusFrame.Status_12_Feedback1, 20, Config.CAN_TIMEOUT_LONG);
+        armTalon.setStatusFramePeriod(StatusFrame.Status_13_Base_PIDF0, 20, Config.CAN_TIMEOUT_LONG);
+
+        // Enable forward soft limit and set the value in encoder ticks
+        armTalon.configForwardSoftLimitEnable(true);
+        armTalon.configForwardSoftLimitThreshold(forwardLimit, Config.CAN_TIMEOUT_LONG);
+
+        // Enable reverse soft limit and set the value in encoder ticks
+        armTalon.configReverseSoftLimitEnable(true);
+        armTalon.configReverseSoftLimitThreshold(reverseLimit, Config.CAN_TIMEOUT_LONG);
+
+        // Max voltage to apply with the talon. 12 is the maximum
+        armTalon.configVoltageCompSaturation(12, Config.CAN_TIMEOUT_LONG);
+        armTalon.enableVoltageCompensation(true);
+
+        // Number of seconds from 0 to full throttle
+        armTalon.configOpenloopRamp(0.6, Config.CAN_TIMEOUT_LONG);
+
+    }
+
+    /**
+     * Initialize the current instance if one does not already exist
+     */
+    public static void init() {
+        if (currentInstance == null) {
+            currentInstance = new Arm();
+        }
+    }
```

Comment:
You could replace all of this init stuff with a simple  
`private static ArmSubsystem INSTANCE = new ArmSubsystem();`  
at the top of your file. They work almost identically, although this method is already used in the intake subsystem and I think we should standardize how we do it.